superset-frontend/src/GlobalStyles.tsx

```diff
@@ -34,6 +34,11 @@ export const GlobalStyles = () => (
       th {
         font-weight: ${theme.typography.weights.bold};
       }
+      // CSS hack to resolve the issue caused by the invisible echart tooltip on
+      // https://github.com/apache/superset/issues/30058
+      .echarts-tooltip[style*='visibility: hidden'] {
+        display: none !important;
+      }
```

Comment:
Couple of options if you want to optimize further:

1) we can probably narrow this from Superset global to being part of the ECharts plugin(s), perhaps using [Emotion Globals](https://emotion.sh/docs/globals).
2) This might be a good excuse to open a PR on ECharts and fix it at the root ;)

 Either way, I don't see this as a blocker for now. Things are lookin' good!

Reply:
Approach 2 is, of course, the fundamental solution, but I decided to leave it as a long-term solution considering the time gap involved in merging the patch and testing Superset compatibility with the corresponding version. (I plan to propose a related PR to echart later.)
Initially, I considered Approach 1, but I had concerns that adding a dependency on @emotion/react to the plugins' npm package might open the possibility for other custom plugins to apply custom global CSS. Therefore, I opted for this approach.